### ISSUE TITLE
[Issue]: Modules won't rebuild on the latest version

### ISSUE THREAD

**Issue by Toxic48** [electron/electron#35869]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

21.0.1

### What operating system are you using?

Windows

### Operating System Version

Windows 10

### What arch are you using?

x64

### Last Known Working Electron version

19.0.10

### Expected Behavior

I am using two electron modules, specifically [electron-bottom-most](https://www.npmjs.com/package/electron-bottom-most) and [electron-disable-minimize](https://www.npmjs.com/package/electron-disable-minimize). When I use "npm start" or rebuild the modules with electron-rebuild the modules rebuild and the program starts normally on Electron v19.0.10.

### Actual Behavior

Since I upgraded to the latest Electron version (v21.0.1) both modules fail to rebuild during the startup and the program won't start. I tried updating node itself, npm, node-gyp, phyton, MVS build tools, and I tried downgrading the Electron version to v18.0.0. When I downgraded the version I got the same results as when I tried using the latest version (both of them failed).

This is what happens when I try to execute "npm start" on the latest version:

   ```
× Preparing native dependencies: 0 / 1
An unhandled error has occurred inside Forge:
node-gyp failed to rebuild 'C:\cwa\node_modules\electron-bottom-most'.
For more information, rerun with the DEBUG environment variable set to "electron-rebuild".

Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1


Error: node-gyp failed to rebuild 'C:\cwa\node_modules\electron-bottom-most'.
For more information, rerun with the DEBUG environment variable set to "electron-rebuild".

Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1


    at NodeGyp.rebuildModule (C:\cwa\node_modules\electron-rebuild\lib\src\module-type\node-gyp.js:120:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ModuleRebuilder.rebuildNodeGypModule (C:\cwa\node_modules\electron-rebuild\lib\src\module-rebuilder.js:98:9)
    at async ModuleRebuilder.rebuild (C:\cwa\node_modules\electron-rebuild\lib\src\module-rebuilder.js:128:14)
    at async Rebuilder.rebuildModuleAt (C:\cwa\node_modules\electron-rebuild\lib\src\rebuild.js:149:13)
    at async Rebuilder.rebuild (C:\cwa\node_modules\electron-rebuild\lib\src\rebuild.js:112:17)
    at async C:\cwa\node_modules\@electron-forge\core\dist\util\rebuild.js:30:9
```

### Additional Information

I only have these two modules and electron-forge installed

**Comment by VerteDinde** [comment]
Hey @Toxic48, thanks for reporting this issue! Would it be possible for you to make a small repro of your project using Electron Forge and the two modules you mentioned? I just tried to make a new project with forge and the two modules, but it built on Windows without issue for me on both Electron 18 and Electron 21 - just want to make sure I'm seeing the same thing that you're seeing. 🙇‍♀️

**Comment by Toxic48** [comment]
> Hey @Toxic48, thanks for reporting this issue! Would it be possible for you to make a small repro of your project using Electron Forge and the two modules you mentioned? I just tried to make a new project with forge and the two modules, but it built on Windows without issue for me on both Electron 18 and Electron 21 - just want to make sure I'm seeing the same thing that you're seeing. 🙇‍♀️

Yeah sure I can make one but, how do I install npm packages on the Electron Fiddle?

**Comment by VerteDinde** [comment]
You can install the modules on Fiddle using the "Modules" sidebar. Once those are installed, you can export your project to a new Forge project by clicking "Make installers for Fiddle" - this was how I tried to repro your issue initially.

![Screen Shot 2022-10-05 at 11 17 38 AM](https://user-images.githubusercontent.com/4624411/194133312-ed4c2417-fb23-48c2-8236-e83c56ed7234.png)

**Comment by Toxic48** [comment]
@VerteDinde okay that screenshot was helpful. Either way I get the same issue. What I did is:
1. Added both of my modules into the modules sidebar
2. Clicked "Tasks" -> "Make installers for Fiddle..." and that worked without errors (that option doesn't rebuild the modules)
3. I ran the Fiddle and it started to rebuild the modules, and then it failed just like in my initial issue

*Theres no need to post my code as a gist because it's the default code, with added modules
![Capture](https://user-images.githubusercontent.com/77108700/194139291-87179637-9654-418f-ab29-24fad395b717.PNG)

The console when I ran the Fiddle:
```
- Checking your system
√ Checking your system
- Resolving Forge Config
√ Resolving Forge Config
- Preparing to Package Application for arch: x64
√ Preparing to Package Application for arch: x64
- Preparing native dependencies
× Preparing native dependencies: 0 / 1

An unhandled error has occurred inside Forge:
node-gyp failed to rebuild 'C:\Users\pc\AppData\Local\Temp\electron-packager\win32-x64\black-mask-tap-rcd6l-win32-x64\resources\app\node_modules\electron-bottom-most'.
For more information, rerun with the DEBUG environment variable set to "electron-rebuild".

Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1


Error: node-gyp failed to rebuild 'C:\Users\pc\AppData\Local\Temp\electron-packager\win32-x64\black-mask-tap-rcd6l-win32-x64\resources\app\node_modules\electron-bottom-most'.
For more information, rerun with the DEBUG environment variable set to "electron-rebuild".

Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1


    at NodeGyp.rebuildModule (C:\Users\pc\AppData\Local\Temp\tmp-4356-T79EaqNmuXnM\node_modules\electron-rebuild\lib\src\module-type\node-gyp.js:120:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ModuleRebuilder.rebuildNodeGypModule (C:\Users\pc\AppData\Local\Temp\tmp-4356-T79EaqNmuXnM\node_modules\electron-rebuild\lib\src\module-rebuilder.js:98:9)
    at async ModuleRebuilder.rebuild (C:\Users\pc\AppData\Local\Temp\tmp-4356-T79EaqNmuXnM\node_modules\electron-rebuild\lib\src\module-rebuilder.js:128:14)
    at async Rebuilder.rebuildModuleAt (C:\Users\pc\AppData\Local\Temp\tmp-4356-T79EaqNmuXnM\node_modules\electron-rebuild\lib\src\rebuild.js:149:13)
    at async Rebuilder.rebuild (C:\Users\pc\AppData\Local\Temp\tmp-4356-T79EaqNmuXnM\node_modules\electron-rebuild\lib\src\rebuild.js:112:17)
    at async C:\Users\pc\AppData\Local\Temp\tmp-4356-T79EaqNmuXnM\node_modules\@electron-forge\core\dist\util\rebuild.js:30:9
```

**Comment by VerteDinde** [comment]
@oalfroukh @Toxic48 Ah interesting, thanks! It looks like we may need a different variation of the flag for Windows builds 🤔 Will take a look and get this fixed, thanks for both for the reports!

**Comment by ludvigintel** [comment]
@VerteDinde Hi, do you have any updates?

**Comment by oalfroukh** [comment]
Hi @VerteDinde 
Hope you're well!
Any update about `/std:c++17` issue in `electron-rebuild`? we are waiting for your fix to upgrade Electron in our app to the latest version 
Thanks in advance!

**Comment by arakawayasuaki** [comment]
Hi,

I was able to build a project with v21 that included robotjs with the following specification

```
  ["OS=='win'", {
                'defines': ['IS_WINDOWS'],
                "msvs_settings": {
                    "VCCLCompilerTool": {
                        "AdditionalOptions": ["-std:c++17", ],
                    },
                },
 }]
```

**Comment by ludvigintel** [comment]
> Hi,
> 
> I was able to build a project with v21 that included robotjs with the following specification
> 
> ```
>   ["OS=='win'", {
>                 'defines': ['IS_WINDOWS'],
>                 "msvs_settings": {
>                     "VCCLCompilerTool": {
>                         "AdditionalOptions": ["-std:c++17", ],
>                     },
>                 },
>  }]
> ```

Did you add the that specification for a custom binding.gyp in the root of your project? And did the apply for all binding.gyp in node_modules?

**Comment by oalfroukh** [comment]
Hi All,
The issue is fixed in Electron `v22.0.0-beta.6` https://github.com/electron/electron/pull/36369, we are waiting to fix it in `v20 & v21`

Thanks for [deepak1556](https://github.com/deepak1556)

**Comment by ludvigintel** [comment]
Awesome news! @deepak1556 when will this be released in v21?
